File: WorldConquest/Models/GameState.cs
using System;
using System.Collections.Generic;

namespace WorldConquest.Models
{
    public class GameState
    {
        public string GameId { get; set; }
        public List<Player> Players { get; set; }
        public List<Country> Countries { get; set; }
        public List<Province> Provinces { get; set; }
        public List<DiplomaticRelation> DiplomaticRelations { get; set; }
        public DateTime CurrentDate { get; set; }
        public string CurrentTurnPlayerId { get; set; }
        public List<Army> Armies { get; set; }
    }
}

File: WorldConquest/client/src/models/GameState.ts
import { Player } from "./Player";
import { Country } from "./Country";
import { Province } from "./Province";
import { DiplomaticRelation } from "./DiplomaticRelation";
import { Army } from "./Army";

export interface GameState {
  gameId: string;
  players: Player[];
  countries: Country[];
  provinces: Province[];
  diplomaticRelations: DiplomaticRelation[];
  currentDate: string; // ISO date string
  currentTurnPlayerId: string;
  armies: Army[];
}